frontend/shared/components/src/events/composables/useEventPermissions.ts

```diff
@@ -0,0 +1,84 @@
+import { AccessRight, NamedObject, Organization, OrganizationTag, PermissionsResourceType } from '@stamhoofd/structures';
+import { useAuth, usePlatform } from '../../hooks';
+
+export function useEventPermissions() {
+    const auth = useAuth();
+    const platform = usePlatform();
+    const permissions = auth.permissions;
+
+    function canAdminSome() {
+        if (!permissions) {
+            return false;
+        }
+
+        return permissions.hasAccessRightForSomeResource(PermissionsResourceType.OrganizationTags, AccessRight.EventWrite) || permissions.hasAccessRightForSomeResource(PermissionsResourceType.Groups, AccessRight.EventWrite);
+    }
+
+    function canAdminAllGroupEvents() {
+        if (!permissions) {
+            return false;
+        }
+        return permissions.hasAccessRightForAllResourcesOfType(PermissionsResourceType.Groups, AccessRight.EventWrite);
+    }
+
+    function canAdminAllTagEvents() {
+        if (!permissions) {
+            return false;
+        }
+        return permissions.hasAccessRightForAllResourcesOfType(PermissionsResourceType.OrganizationTags, AccessRight.EventWrite);
+    }
+
+    function hasGroupRestrictions() {
+        if (!permissions) {
+            return false;
+        }
+
+        if (canAdminAllGroupEvents()) {
+            return false;
+        }
+
+        return permissions.hasAccessRightForSomeResource(PermissionsResourceType.Groups, AccessRight.EventWrite);
+    }
+
+    function hasTagRestrictions() {
+        if (!permissions) {
+            return false;
+        }
+
+        if (canAdminAllTagEvents()) {
+            return false;
+        }
+
+        return permissions.hasAccessRightForSomeResource(PermissionsResourceType.OrganizationTags, AccessRight.EventWrite);
```

Comment:
Dus als je geen toegang hebt tot geen enkele tag, dan is `hasTagRestrictions` true? Waarom is dit - de naam maakt het niet duidelijk - voeg eventueel comments toe om te verduidelijken waarom je iets doet